packages/core/core/src/requests/AssetGraphRequest.js

```diff
@@ -747,110 +889,150 @@ export class AssetGraphBuilder {
         } else if (childNode.type === 'dependency') {
           childDirty = childNode.usedSymbolsDownDirty;
         }
-        if (!visited.has(child) || childDirty) {
+        if (childDirty) {
```

Comment:
Does this mean nodes may be visited more than once now?

Reply:
Unfortunately yes, I've added another comment: https://github.com/parcel-bundler/parcel/blob/d7d0d01a76ccf74c1b2457ce3de1fba297419e89/packages/core/core/src/requests/AssetGraphRequest.js#L846-L863

(But the hope is that this won't happen often. If it does become a problem, then we should tweak the traversal order)